backend/src/config/color.rs

```diff
@@ -11,16 +11,25 @@ pub(crate) struct Color {
 }
 
 impl Color {
-    /// Returns either "black" or "white", depending on which of those has a
-    /// higher contrast to `self`.
+    /// Calculates relative luminance and contrast ratio of `self` to white,
+    /// based on WCAG2.1 Technique G18 (https://www.w3.org/WAI/WCAG21/Techniques/general/G18.html).
+    /// Returns either "black" or "white", depending on contrast ratio.
     pub(crate) fn bw_contrast(self) -> &'static str {
-        let luminance = (self.r as f32 / 255.0) * 0.299
-            + (self.g as f32 / 255.0) * 0.587
-            + (self.b as f32 / 255.0) * 0.114;
+        fn linear_value(color_component: u8) -> f32 {
+            let s_rgb = (color_component as f32) / 255.0;
+            if s_rgb <= 0.04045 {
+                s_rgb / 12.92
+            } else {
+                ((s_rgb + 0.055) / 1.055).powf(2.4)
+            }
+        }
+
+        let relative_luminance = 0.2126 * linear_value(self.r)
+            + 0.7152 * linear_value(self.g)
+            + 0.0722 * linear_value(self.b);
+        let contrast_ratio = 1.05 / (relative_luminance + 0.05);
 
-        // The threshold of 0.6 is fairly arbitrary, but works well in practice.
-        // You will find various thresholds in the internet.
-        if luminance > 0.6 {
+        if contrast_ratio < 4.5 {
             "black"
```

Comment:
So you are calculating the contrast with white and then check if it meets the contrast requirement. If not, we use black. But is 4.5 really the "midpoint", where black and white would have a contrast of 4.5? Because what we want is to chose the color that has the higher contrast, right? So this code seems to imply that if the contrast with white is below 4.5, then the contrast with black is above 4.5. Is that the case? First I wanted to write this comment as "I'm fairly certain this is not the case", but playing with the contrast checker indeed makes it seem like it is. Do you have some source on that?

Otherwise I would probably have written it sth like this:

```
let contrast_with_white = 1.05 / (relative_luminance + 0.05);
let contrast_with_black = 0.05 / (relative_luminance + 0.05);
if contrast_with_white > contrast_with_black ....
```

So I'd say: either write it like that or add a comment explaining why the above assumption holds and why this ensures we use the color with the higher contrast.

Reply:
Oh right, I chose 4.5 here because a ratio below that would not pass accessibility checks. But I'm not 100% sure that necessarily means that black would have a higher ratio, so I believe your suggestion to choose the higher ratio of both makes more sense.